pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java

```diff
@@ -84,17 +94,12 @@ public String getData(
 
     ZNRecord znRecord = _pinotHelixResourceManager.readZKData(path);
     if (znRecord != null) {
-      byte[] serializeBytes = _znRecordSerializer.serialize(znRecord);
-      if (GZipCompressionUtil.isCompressed(serializeBytes)) {
-        try {
-          serializeBytes = GZipCompressionUtil.uncompress(new ByteArrayInputStream(serializeBytes));
-        } catch (IOException e) {
-          throw new RuntimeException(e);
-        }
-      }
+      byte[] serializeBytes = serialize(znRecord);
       return new String(serializeBytes, StandardCharsets.UTF_8);
```

Comment:
Remove `serialize()`
```suggestion
      try {
        return MAPPER.writeValueAsString(znRecords);
      } catch (IOException e) {
        throw new RuntimeException(e);
      }
```